source/server/listener_manager_impl.cc

```diff
@@ -622,6 +622,20 @@ void ListenerManagerImpl::stopListeners() {
   }
 }
 
+void ListenerManagerImpl::closeListeners() {
+  for (auto& listener : active_listeners_) {
+    listener->close();
```

Comment:
does this ultimately mutate active_listeners_?  When does a previously active listener get removed from the collection?

Reply:
Listeners are only removed when LDS removes them. This call does not mutate the vector (or list?), only the elements inside.

Reply:
close() maybe should be renamed to closeSocket() to make it clearer.